### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by SweetOps**
## What
* Make bucket creation optional
* Support s3 websites
* Fixed mistakes

### CODE REVIEW

**Comment by SweetOps** [comment]
screenshot 2017-09-15 15 54 49

File: main.tf [hunk 1]
```diff
@@ -12,33 +12,39 @@ resource "aws_cloudfront_origin_access_identity" "default" {
...
-    principals {
-      type        = "AWS"
-      identifiers = ["${aws_cloudfront_origin_access_identity.default.iam_arn}"]
-    }
-  }
-
-  statement {
-    actions   = ["s3:ListBucket"]
-    resources = ["arn:aws:s3:::${aws_s3_bucket.origin.bucket}"]
+data "template_file" "bucket_policy_file" {
```
Comment (osterman) [inline main.tf]: Do not see the value proposition of using template file over a policy document. Template file does not validate the JSON while the the policy document does.

File: main.tf [hunk 2]
```diff
@@ -124,14 +130,12 @@ resource "aws_cloudfront_distribution" "default" {
...
 
   tags = "${module.distribution_label.tags}"
-
-  depends_on = ["aws_s3_bucket.logs"]
 }
 
 module "dns_aliases" {
-  source          = "git::https://github.com/cloudposse/tf_vanity?ref=tags/0.2.0"
+  source          = "git::https://github.com/cloudposse/tf_vanity.git?ref=generalize"
   aliases         = ["${var.aliases}"]
```
Comment (osterman) [inline main.tf]: Aliases is already a list. This is double listing it, no?

File: main.tf [hunk 3]
```diff
@@ -12,33 +12,39 @@ resource "aws_cloudfront_origin_access_identity" "default" {
...
 
 resource "aws_s3_bucket" "origin" {
+  count  = "${signum(length(var.custom_bucket_domain_name)) == 1 ? 0 : 1}"
   bucket = "${module.origin_label.id}"
   acl    = "private"
-  policy = "${data.aws_iam_policy_document.origin.json}"
+  policy = "${data.template_file.bucket_policy.rendered}"
   tags   = "${module.origin_label.tags}"
+
+  website {
```
Comment (osterman) [inline main.tf]: No, cannot use `website` with s3 origins

File: main.tf [hunk 4]
```diff
@@ -12,33 +12,39 @@ resource "aws_cloudfront_origin_access_identity" "default" {
...
+    bucket_name = "${module.origin_label.id}"
   }
 }
 
 resource "aws_s3_bucket" "origin" {
+  count  = "${signum(length(var.custom_bucket_domain_name)) == 1 ? 0 : 1}"
   bucket = "${module.origin_label.id}"
   acl    = "private"
-  policy = "${data.aws_iam_policy_document.origin.json}"
+  policy = "${data.template_file.bucket_policy.rendered}"
```
Comment (osterman) [inline main.tf]: we should use `aws_s3_bucket_policy` instead

**Pull request merged**